vignettes/savvyr_example.Rmd

```diff
@@ -0,0 +1,114 @@
+---
+title: "Estimation of AE probabilities with savvyr"
+package: savvyr
+bibliography: "../inst/REFERENCES.bib"
+output:
+  rmarkdown::html_vignette:
+          toc: true
+vignette: |
+  %\VignetteEncoding{UTF-8}
+  %\VignetteIndexEntry{Estimation of AE probabilities with savvyr}
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options:
+  chunk_output_type: console
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+library(savvyr)
+library(kableExtra)
+```
+
+
+
+
+# Example using dummy data
+
+We generate the dataset $S1$ in  @stegherr_meta_analytic_2021 using the parameter values for Arm A:
+
+```{r, include=TRUE, echo=TRUE}
+# sample size
+N <- 200
+
+# support of uniform censoring distribution
+min.cens <- 0
+max.cens <- 1000
+
+# hazards for the three event types
+haz.AE <- 0.00265
+haz.death <- 0.00151
+haz.soft <- 0.00227
+
+
+
+# generate dataset
+set.seed(2020)
+dat1 <- generate_data(N, cens = c(min.cens, max.cens), haz.AE, haz.death, haz.soft)
+
+
+# compute tau
+tau <- max(dat1[, "time_to_event"])
+```
+
+The structure of the dataset looks as follows:
+
+```{r, include=TRUE, echo=TRUE}
+kable(head(dat1, 10), align = c("crcr"))
+```
+
+For this dataset we then compute all the estimators used in the comparisons in @stegherr_survival_2021 and @stegherr_estimating_2021:
+
+```{r, include=TRUE, echo=TRUE}
+# compute each estimator
+IP <- inc_prop(dat1, tau)
+ID <- prop_trans_inc_dens(dat1, tau)
+KM <- one_minus_kaplan_meier(dat1, tau)
+
+# competing event "death only"
+IDCE2 <- prop_trans_inc_dens_ce(dat1, ce = 2, tau)
+AJ2 <- aalen_johansen(dat1, ce = 2, tau)
+
+# account for all competing events
+IDCE3 <- prop_trans_inc_dens_ce(dat1, ce = 3, tau)
+AJ3 <- aalen_johansen(dat1, ce = 3, tau)
+
+# display
+tab <- rbind(IP, ID, KM, IDCE2, AJ2[1:2], IDCE3, AJ3[1:2])
+colnames(tab) <- c("estimated AE probability", "variance of estimation")
+rownames(tab) <- c(
+  "incidence proportion", "probability transform incidence density ignoring competing event",
+  "1 - Kaplan-Meier", "probability transform incidence density (death only)",
+  "Aalen-Johansen (death only), AE risk", "probability transform incidence density (all CEs)",
+  "Aalen-Johansen (all CEs), AE risk"
+)
+
+# probability of AE
```

Comment:
generally I would try to move as many (if not all) comments from code into actual text with sentences. This helps the reader a lot to understand the whole workflow better.

Reply:
adapted